### PR TITLE
Lot of refactoring

### PR DESCRIPTION
**Pull request by 907th**
Hi! I really like your gem and have used it a long time.

I made lot of refactoring in this PR:
- Add support for custom exception classes/objects to `#assert` (e.g. `assert false, MyError.new('message')`);
- Allow to enabled/disable assertions many times (`SolidAssert.disable_assertions` method added);
- Update dependencies, remove jeweler gem (there are identical rake helpers), use newer RSpec syntax in tests.

PS. I also updated gem version var to `0.8.0` but I'm not sure if my refactoring not breaking compatibility.

I'm waiting for your response. Thanks!

### CODE REVIEW

**Comment by jorgemanrubia** [comment]
Hi @907th,

Thanks a lot, really appreciated!

Would you be interested in having permissions on the repo? I could use some help to maintain this gem.

Travis seems to be failing due to wrong bundler version.

Regards

**Comment by 907th** [comment]
@jorgemanrubia Thank you for so fast response! Yes, I can help with maintaining it! Thank you.

**Comment by jorgemanrubia** [comment]
You are a collaborator now :)! Thanks Alex

**Comment by 907th** [comment]
@jorgemanrubia I've fixed Travis build (added `.travis.yml`). But I can not release the gem to Rubygems:
> Pushing gem to https://rubygems.org...
You do not have permission to push to this gem.

Could you please give me a permission to release the gem?

**Comment by jorgemanrubia** [comment]
Done @907th! Added permissions to the email that is shown in your GH profile (@pisem.net)

**Comment by 907th** [comment]
@jorgemanrubia Thank you! Released v0.8.0.

File: bin/console [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env ruby
+
+require "bundler/setup"
+require "solid_assert"
+
+require "irb"
+IRB.start
```
Comment (joshuaflanagan) [inline bin/console]: I know this PR is already merged, but as long as you are cleaning/updating things, you might as well get rid of this script. The same functionality is already available via `bundle console` (no need to create your own `bin/console` anymore).

Comment (907th) [inline bin/console]: @joshuaflanagan I didn't know about that feature. Thank you!

**Pull request merged**